internal/adapters/config/mongodb.go

```diff
@@ -0,0 +1,29 @@
+package config
+
+import (
+	"context"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+	"log/slog"
+)
+
+func Connect(db DB) (*mongo.Database, error) {
+	slog.Info("Connecting to mongodb database")
+	opts := options.Client().ApplyURI(db.Uri).SetAuth(
+		options.Credential{
+			Username: db.User,
+			Password: db.Password,
+		})
+	conn, err := mongo.Connect(context.TODO(), opts)
```

Comment:
You application should feed the main context to the mongo connection so that if something interrupts the app this connection gets properly closed
Note that this should not the API context, but the application context

Reply:
I added the app context and the database disconnection inside the graceful shutdown.